go/vt/vttablet/tabletmanager/vreplication/controller_plan_test.go

```diff
@@ -21,35 +21,54 @@ import (
 	"testing"
 )
 
+type testControllerPlan struct {
+	query        string
+	opcode       int
+	numInserts   int
+	selector     string
+	applier      string
+	delCopyState string
+}
+
 func TestControllerPlan(t *testing.T) {
 	tcases := []struct {
 		in   string
-		plan *controllerPlan
+		plan *testControllerPlan
 		err  string
 	}{{
 		// Insert
 		in: "insert into _vt.vreplication values(null)",
-		plan: &controllerPlan{
-			opcode: insertQuery,
-			query:  "insert into _vt.vreplication values (null)",
+		plan: &testControllerPlan{
+			query:      "insert into _vt.vreplication values(null)",
+			opcode:     insertQuery,
+			numInserts: 1,
 		},
 	}, {
 		in: "insert into _vt.vreplication(id) values(null)",
-		plan: &controllerPlan{
-			opcode: insertQuery,
-			query:  "insert into _vt.vreplication(id) values (null)",
+		plan: &testControllerPlan{
+			query:      "insert into _vt.vreplication(id) values(null)",
+			opcode:     insertQuery,
+			numInserts: 1,
 		},
 	}, {
 		in: "insert into _vt.vreplication(workflow, id) values('', null)",
-		plan: &controllerPlan{
-			opcode: insertQuery,
-			query:  "insert into _vt.vreplication(workflow, id) values ('', null)",
+		plan: &testControllerPlan{
+			query:      "insert into _vt.vreplication(workflow, id) values('', null)",
+			opcode:     insertQuery,
+			numInserts: 1,
+		},
+	}, {
+		in: "insert into _vt.vreplication values(null), (null)",
```

Comment:
why is this now a valid construct? is it that the controller_plan allows it but it will fail at the database level?

Reply:
This test demonstrates multi-row insert, which was previously not possible. As for the content of the row, the plan builder just passes it through. So, this specific construct will fail at the db level.

As you can see all the other tests do the same thing (in terms of content). In a way, theses tests also show what the plan builder does not validate.